virtual_rainforest/models/abiotic/radiation.py

```diff
@@ -0,0 +1,226 @@
+    ) -> None:
+        """Calculates daily net radiation at the forest floor.
+
+        Args:
+            canopy_absorption: NDArray[np.float32]: absorption by canopy layers, [J m-2]
+
+        Returns:
+            self.netradiation_surface: NDArray[np.float32], net shortwave radiation at
+                the forest floor, [J m-2]
+        """
+        self.netradiation_surface = (
+            self.topofcanopy_radiation
+            - canopy_absorption  # np.sum(canopy_absorption, axis=1)  # what axis?
+            - self.longwave_soil
+            - self.longwave_canopy  # np.sum(self.longwave_canopy, axis=1)
+        )
+        """Daily net shortwave radiation at the surface (= forest floor), [J m-2]"""
+
+    def radiation_balance(
+        self,
+        elevation: NDArray[np.float32],
+        shortwave_in: NDArray[np.float32],
+        sunshine_hours: NDArray[np.float32],
+        albedo_vis: NDArray[np.float32],
+        albedo_shortwave: NDArray[np.float32],
+        canopy_temperature: NDArray[np.float32],
+        surface_temperature: NDArray[np.float32],
+        canopy_absorption: NDArray[np.float32],
+    ) -> None:
+        """Calculate radiation balance.
+
+        Args:
+            elevation: NDArray[np.float32], elevation [m]
+            shortwave_in: NDArray[np.float32], daily downward shortwave radiation[J m-2]
+            sunshine_hours: NDArray[np.float32], fraction of sunshine hours, between 0
+                (100% cloud cover) and 1 (cloud free sky)
+            albedo_vis: NDArray[np.float32], visible light albedo, default = 0.03
+            albedo_shortwave: NDArray[np.float32], shortwave albedo, default = 0.17
+            canopy_temperature: NDArray[np.float32], canopy temperature of n layers [C]
+            surface_temperature: NDArray[np.float32], surface soil temperature [C]
+            canopy_absorption: NDArray[np.float32]: absorption by canopy [J m-2]
+
+        Returns:
+            elevation: NDArray[np.float32], elevation [m]
+            topofcanopy_radiation: NDArray[np.float32], top of canopy downward shortwave
+                radiation, [J m-2]
+            ppfd: NDArray[np.float32], top of canopy photosynthetic photon flux density,
+                [mol m-2]
+            longwave_canopy: NDArray[np.float32], longwave radiation from n individual
+                canopy layers [J m-2]
+            longwave_soil: NDArray[np.float32], longwave radiation from soil [J m-2]
+            netradiation_surface: NDArray[np.float32], net shortwave radiation at
+                the surface (=forest floor)[J m-2]
+        """
+
+        tau = self.calc_ppfd(shortwave_in, sunshine_hours, albedo_vis)
+        self.calc_topofcanopy_radiation(
+            tau, shortwave_in, sunshine_hours, albedo_shortwave
+        )
+        self.calc_longwave_radiation(canopy_temperature, surface_temperature)
+        self.calc_netradiation_surface(canopy_absorption)
+        del tau
```

Comment:
I'm not sure if the `del` is needed here? I don't think it is wrong, but I think `tau` will just be garbage collected when the method returns.